lib/cgrates_web_jsonapi/cdrs/crd_filter.ex

```diff
@@ -0,0 +1,66 @@
+defmodule CgratesWebJsonapi.Cdrs.CdrFilter do
+  @moduledoc """
+  CDRs filters
+  """
+
+  import Ecto.Query
+
+  defmacro __using__(_opts) do
+    quote do
+      def apply_filter(query, filter) do
+        filter
+        |> Map.to_list()
+        |> Enum.reduce(query, fn {key, value}, q -> filter(q, key, value) end)
+      end
+
+      def filter(q, "created_at_lte", val), do: q |> where([r], r.created_at <= ^val)
```

Comment:
In order to reuse filter in the JaResource controller and the  cdrs context `filter` function must have the same interface `filter(_conn, q, filter_name, filter_value)`. Just add the first argument (_conn) to filters by  created_at_lte, created_at_gte,  rating_plan_tag. and send it as nil in `apply_filter`